streamer/periodconcat_node.py

```diff
@@ -0,0 +1,256 @@
+        self.hours: int = hours
+        self.minutes: int = minutes
+        self.seconds: float = seconds
+        
+      @staticmethod
+      def parse(duration: str) -> 'ISO8601':
+        """Converts duration from iso8601 format to native python data-types."""
+        
+        pattern = ('^P(?:([0-9]*)Y)?(?:([0-9]*)M)?(?:([0-9]*)D)?'
+                   '(?:T(?:([0-9]*)H)?(?:([0-9]*)M)?(?:([0-9.]*)S)?)?$')
+        matches = re.search(pattern, duration)
+        assert matches is not None, "ISO8601: Duration parsing failed."
+        
+        groups = [int(match) for match in matches.groups('0')[:5]]
+        seconds = float(matches.groups('0')[5])
+        
+        return ISO8601(groups[0], groups[1], groups[2],
+                       groups[3], groups[4], seconds)
+      
+      def __str__(self) -> str:
+        """Return the duration in iso8601 format."""
+        
+        return "P{}Y{}M{}DT{}H{}M{}S".format(self.years, self.months, self.days,
+                                             self.hours, self.minutes, self.seconds)
+      
+      def __add__(self, other: 'ISO8601') -> 'ISO8601':
+        """Addes two iso8601 time durations together, and returns their sum.
+        Since each time container is not limited by when it overflows (ie, we can have 61S or 36H
+        this format doesn't limit us to 59.99S or 23H), we can perform a simple addition between
+        matching containers."""
+        
+        return ISO8601(
+          self.years + other.years,
+          self.months + other.months,
+          self.days + other.days,
+          self.hours + other.hours,
+          self.minutes + other.minutes,
+          self.seconds + other.seconds)
+      
+      def __iadd__(self, other: 'ISO8601') -> 'ISO8601':
+        return self + other
+    
+    dash_namespace = 'urn:mpeg:dash:schema:mpd:2011'
+    
+    def find(elem: ElementTree.Element, *args: str) -> ElementTree.Element:
+      """A better interface for the Element.find() method.
+      Use it only if it is guaranteed that the element we are searching for is inside,
+      Otherwise it will raise an AssertionError."""
+      full_path = ''
+      for tag in args:
+        full_path+='{'+dash_namespace+'}'+tag+'/'
+      child_elem =  elem.find(full_path[:-1])
+      # elem.find() returns either an ElementTree.Element or None.
+      assert child_elem is not None
+      return child_elem
+    
+    def findall(elem: ElementTree.Element, *args: str) -> List[ElementTree.Element]:
+      """A better interface for the Element.findall() method"""
+      full_path = ''
+      for tag in args:
+        full_path+='{'+dash_namespace+'}'+tag+'/'
+      return elem.findall(full_path[:-1])
+    
+    # We will add the time of each period on this container to set that time later
+    # as 'mediaPresentationDuration' for the MPD tag for VOD.
+    # For LIVE we will use this object to determine the start time of the next period.
+    total_time: ISO8601 = ISO8601()
```

Comment:
You don't need to do this.  If you move the mediaPresentationDuration of each mpd into the Period element's duration attribute, you can omit mediaPresentationDuration from the combined mpd element.  Players must be able to handle this, so you can skip all the duration parsing above.